### PR TITLE
Added XBackBone

### PR DESCRIPTION
**Pull request by angrycuban13**
# Pull request

**Purpose**
Adding xbackbone.
**Open Questions and Pre-Merge TODOs**
Check all boxes as they are completed

- [x] Use github checklists. When solved, check the box and explain the answer.

**Requirements**
Check all boxes as they are completed

- [x] These changes meet the standards for [contributing](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CONTRIBUTING.md).
- [x] I have read the [code of conduct](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: compose/.apps/xbackbone/xbackbone.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      driver: json-file
+      options:
+        max-file: ${DOCKERLOGGING_MAXFILE}
+        max-size: ${DOCKERLOGGING_MAXSIZE}
+    restart: unless-stopped
+    volumes:
+      - /etc/localtime:/etc/localtime:ro
+      - ${DOCKERCONFDIR}/xbackbone:/app/resources/database
+      - ${DOCKERCONFDIR}/xbackbone/logs:/app/logs
+      - ${XBACKBONE_UPLOADDIR}/images:/app/storage
```
[the remark this answers is not in the document]
Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml:21]: How would we do the labels? Right now I have it set to

` com.dockstarter.appvars.xbackbone_uploaddir: "/mnt/downloads"`

Leave it empty, or are you saying just kind of let people decide where the directories go and not necessarily force it to `/mnt/downloads/images`?

[the remark this answers is not in the document]
Comment (nemchik) [reply compose/.apps/xbackbone/xbackbone.yml:21]: What I'm getting at is we don't need to let users pick a folder via a variable and then force a subfolder beneath it on their host. If there needs to be separate subfolders inside the container under `/app/storage` we should make variables for all of them, or just do the parent. As for the default value I think I did something similar for nextcloud. Keep the name of the app in the folder. Something like `/mnt/xbackbone`. That's just the default value we'd put in labels, users can change it.

[the remark this answers is not in the document]
Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml:21]: I like that idea. I'll update it.

[the remark this answers is not in the document]
Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml]: 269ee96

File: compose/.apps/xbackbone/xbackbone.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    logging:
+      driver: json-file
+      options:
+        max-file: ${DOCKERLOGGING_MAXFILE}
+        max-size: ${DOCKERLOGGING_MAXSIZE}
+    restart: unless-stopped
+    volumes:
+      - /etc/localtime:/etc/localtime:ro
+      - ${DOCKERCONFDIR}/xbackbone:/app/resources/database
+      - ${DOCKERCONFDIR}/xbackbone/logs:/app/logs
```
Comment (nemchik) [inline compose/.apps/xbackbone/xbackbone.yml:20]: Missing shared directory. Otherwise this looks great.

Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml:20]: 73f49cb

Comment (nemchik) [reply compose/.apps/xbackbone/xbackbone.yml:20]: Sort alphabetically :rofl:

Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml:20]: God dammit... lol

Comment (angrycuban13) [reply compose/.apps/xbackbone/xbackbone.yml:20]: 71ad4ce 🤞

**Pull request merged**